viadot/utils.py

```diff
@@ -83,29 +109,37 @@ def handle_api_response(
     params: Dict[str, Any] = None,
     headers: Dict[str, Any] = None,
     timeout: tuple = (3.05, 60 * 30),
+    method: Literal["GET", "POST", "DELETE"] = "GET",
+    data: str = None,
 ) -> requests.models.Response:
     """
     Handle an HTTP response by applying retries and handling some common response
     codes.
 
     Args:
-        url (str): the URL which trying to connect.
-        auth (tuple, optional): authorization information. Defaults to None.
-        params (Dict[str, Any], optional): the request parameters. Defaults to None.
-        headers: (Dict[str, Any], optional): the request headers. Defaults to None.
-        timeout (tuple, optional): the request times out. Defaults to (3.05, 60 * 30).
+        url (str): The URL which trying to connect.
+        auth (tuple, optional): Authorization information. Defaults to None.
```

Comment:
1) typo: it's authentication, not authorization
2) I would link here to request docs as it's unclear here what input is expected. See https://requests.readthedocs.io/en/latest/user/authentication/ for all possible auth types. As you can see, it's not just `tuple` (basic auth), but it can by any class that handles auth, some of which are provided by `requests`. And if we only support basic auth, this should be stated here, probably with an example (`(username, password)`).

Reply:
[📝 Updated docstrings in utils.py](https://github.com/dyvenia/viadot/pull/661/commits/ec9e1879f7155aa71233f9fad2aeba5560f290c0)